hash_set/hash_set.mbt

```diff
@@ -0,0 +1,562 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// This module implements mutable set as hashtable.
+
+/// MutableSet is an open-addressing hashset based on robin hood hash.
+pub struct MutableSet[T] {
+  priv mut directory : @vec.Vec[Option[Item[T]]]
```

Comment:
Also it may not be a good idea to use `Vec` here. The `Vec` automatically grows, but you will need to reallocate and rehash anyway, so using `Vec` will cause an extra boxing.

Reply:
Should I use Array directly to implement it (and then implement expansion-related operations myself)? The performance of List in terms of indexing is not very good.

Reply:
Yes! the expansion operation of directory will only appear in with_capacity. It is completely reasonable to use Array. I have changed it to Array (fe3b263).